ext/pdo_odbc/config.m4

```diff
@@ -71,22 +70,54 @@ if test "$PHP_PDO_ODBC" != "no"; then
         ;;
   esac
 
-  if test -n "$pdo_odbc_dir"; then
-    PDO_ODBC_INCDIR="$pdo_odbc_dir/include"
-    PDO_ODBC_LIBDIR="$pdo_odbc_dir/$PHP_LIBDIR"
+  if test ! -z "$pdo_odbc_pkgconfig_module"; then
+    AC_MSG_RESULT([$pdo_odbc_flavour using pkg-config])
+    PKG_CHECK_MODULES([PDO_ODBC], [$pdo_odbc_pkgconfig_module])
+    PHP_EVAL_INCLINE([$PDO_ODBC_CFLAGS])
+    PHP_EVAL_LIBLINE([$PDO_ODBC_LIBS], [PDO_ODBC_SHARED_LIBADD])
+    PDO_ODBC_INCLUDE="$PDO_ODBC_CFLAGS -DPDO_ODBC_TYPE=\\\"$pdo_odbc_flavour\\\""
   else
-    PDO_ODBC_INCDIR="$pdo_odbc_def_incdir"
-    PDO_ODBC_LIBDIR="$pdo_odbc_def_libdir"
-  fi
-
-  AC_MSG_RESULT([$pdo_odbc_flavour
-          libs       $PDO_ODBC_LIBDIR,
-          headers    $PDO_ODBC_INCDIR])
-
-  if test ! -d "$PDO_ODBC_LIBDIR" ; then
-    AC_MSG_WARN([library dir $PDO_ODBC_LIBDIR does not exist])
+    if test -n "$pdo_odbc_dir"; then
+      PDO_ODBC_INCDIR="$pdo_odbc_dir/include"
+      PDO_ODBC_LIBDIR="$pdo_odbc_dir/$PHP_LIBDIR"
+    else
+      PDO_ODBC_INCDIR="$pdo_odbc_def_incdir"
+      PDO_ODBC_LIBDIR="$pdo_odbc_def_libdir"
+    fi
+
+    AC_MSG_RESULT([$pdo_odbc_flavour
+            libs       $PDO_ODBC_LIBDIR,
+            headers    $PDO_ODBC_INCDIR])
+
+    if test ! -d "$PDO_ODBC_LIBDIR" ; then
+      AC_MSG_WARN([library dir $PDO_ODBC_LIBDIR does not exist])
+    fi
+
+    PDO_ODBC_INCLUDE="$pdo_odbc_def_cflags -I$PDO_ODBC_INCDIR -DPDO_ODBC_TYPE=\\\"$pdo_odbc_flavour\\\""
```

Comment:
```suggestion
    PDO_ODBC_INCLUDE="$pdo_odbc_def_cflags -I$PDO_ODBC_INCDIR -DPDO_ODBC_TYPE=\"$pdo_odbc_flavour\""
```

Reply:
This needs to be rechecked a bit if it is inserted in C correctly because shell emits some warnings `warning: missing terminating` in config.log.